pkg/agent/controller/networkpolicy/l7engine/l7visbility_watcher.go

```diff
@@ -0,0 +1,393 @@
+					direction = v1alpha2.DirectionEgress
+				case "Both":
+					direction = v1alpha2.DirectionBoth
+				}
+				for _, node := range nodeList.Items {
+					err = l7w.startSuricataAndCreateTC(direction, node.Name, nil, namespaceLabels)
+					if err != nil {
+						klog.ErrorS(err, "Suricata  or TrafficControl didnt initialize")
+						return
+					}
+				}
+			} else {
+				err = l7w.deleteTC(nil, namespaceLabels)
+				if err != nil {
+					klog.ErrorS(err, "Failed to delete Traffic Control")
+					return
+				}
+			}
+		} else if event.Type == watch.Deleted {
+			err = l7w.deleteTC(nil, namespaceLabels)
+			if err != nil {
+				klog.ErrorS(err, "Failed to delete Traffic Control")
+				return
+			}
+		}
+	}
+}
+
+func (l7w *l7Watcher) StartSuricataIfNotRunning(nodeName string) error {
+	L7mutex.Lock()
+	defer L7mutex.Unlock()
+
+	// Get the node by name
+	node, err := l7w.clientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		klog.ErrorS(err, "")
+		return err
+	}
+
+	// Check if Suricata is running on the node
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionTrue {
+			l7w.checkSuricataProcess(nodeName)
+		}
+	}
+	return nil
+}
+
+func (l7w *l7Watcher) checkSuricataProcess(nodeName string) error {
+	currentnode := os.Getenv("NODE_NAME")
+	if nodeName == currentnode {
+		cmd := exec.Command("bash", "-c", "ps aux | grep suricata")
+		stdout, err := cmd.CombinedOutput()
+		if err != nil {
+			return err
+		}
+		if !strings.Contains(string(stdout), "suricata -c") {
+			l7w.l7Reconciler.startSuricata()
+		}
+	}
+	return nil
+}
+
+func createTrafficControl(
+	tcName string,
+	nsSelector,
+	podSelector map[string]string,
+	direction v1alpha2.Direction,
+	action v1alpha2.TrafficControlAction,
+	targetPort interface{},
+	tcController *trafficcontrol.Controller) error {
+	tc := &v1alpha2.TrafficControl{
+		ObjectMeta: metav1.ObjectMeta{Name: tcName},
+		Spec: v1alpha2.TrafficControlSpec{
+			Direction: direction,
+			Action:    action,
+		},
+	}
+	if nsSelector != nil {
+		tc.Spec.AppliedTo.NamespaceSelector = &metav1.LabelSelector{MatchLabels: nsSelector}
+	}
+	if podSelector != nil {
+		tc.Spec.AppliedTo.PodSelector = &metav1.LabelSelector{MatchLabels: podSelector}
+	}
+	tc.Spec.TargetPort.OVSInternal = targetPort.(*v1alpha2.OVSInternalPort)
+	trafficControlApi := tcController.CrdClient.CrdV1alpha2().TrafficControls()
+	_, err := trafficControlApi.Create(context.TODO(), tc, metav1.CreateOptions{})
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		klog.ErrorS(err, "Failed to create TrafficControl")
+		return err
+	}
+	return nil
+}
+
+func (l7w *l7Watcher) postTrafficControl(tcName string, direction v1alpha2.Direction, podLabels, namespaceLabels map[string]string) error {
+	targetPort := &v1alpha2.OVSInternalPort{Name: config.L7NetworkPolicyTargetPortName}
+	var err error
+	if len(podLabels) != 0 {
```

Comment:
It doesn't make sense to use `if else` here on `len(podLabels) != 0`, but then use `if else` in `createTrafficControl` on `podSelector != nil`.